cockatrice/src/zoneviewzone.cpp

```diff
@@ -102,18 +103,52 @@ void ZoneViewZone::reorganizeCards()
     CardList cardsToDisplay(cards);
     if (sortByName || sortByType)
         cardsToDisplay.sort((sortByName ? CardList::SortByName : 0) | (sortByType ? CardList::SortByType : 0));
+
+    int typeColumn = 0;
+    int longestRow = 0;
+    if (pileView && sortByType) // we need sort by type enabled for the feature to work
+        setPileViewPositions(cardCount, cardsToDisplay, typeColumn, longestRow);
+    else {
+        for (int i = 0; i < cardCount; i++) {
+            CardItem *c = cardsToDisplay.at(i);
+            qreal x = (i / rows) * CARD_WIDTH;
+            qreal y = (i % rows) * CARD_HEIGHT / 3;
+            c->setPos(x + 5, y + 5);
+            c->setRealZValue(i);
+        }
+    }
     
+    if (pileView && sortByType)
+        optimumRect = QRectF(0, 0, qMax(typeColumn + 1, 3) * CARD_WIDTH + 20, ((longestRow - 1) * CARD_HEIGHT) / 3 + CARD_HEIGHT + 60);
+    else 
+        optimumRect = QRectF(0, 0, qMax(cols, 1) * CARD_WIDTH + 20, ((rows - 1) * CARD_HEIGHT) / 3 + CARD_HEIGHT + 20);
```

Comment:
Can you factor out the common parts of the two branches here? Have each parameter of `QRectF` named, and initialize them in the respective branches?